litgpt/__init__.py

```diff
@@ -16,4 +18,12 @@
 logging.getLogger("torch.distributed.fsdp._optim_utils").disabled = True
 logging.getLogger("torch.distributed.fsdp._debug_utils").disabled = True
 
+supported, version = check_library_version("bitsandbytes", "0.42.0")
+if version and not supported:
+    warnings.warn(
+        f"You have bitsandbytes v{version} installed but "
+        "LitGPT only supports bitsandbytes v0.42.0. "
+        "This may result in errors when using quantization."
+    )
```

Comment:
This warning will be a false-positive in cases where you're not using bnb. Maybe you've installed with `[all]` for a different reason.

I think if anything, this should be checked only in the case where quantization is requested.

Reply:
Ok that's fair. I wanted to avoid having it in too many places in the code but I'm happy to change it. Thanks for referencing the 

```
from lightning_utilities.core.imports import RequirementCache
```

Didn't know about this, and this will come in handy.

Reply:
> I wanted to avoid having it in too many places

You could have put it in a function to reuse it to avoid the duplication.